apps/wallet/src/ui/app/pages/swap/SwapPageForm.tsx

```diff
@@ -0,0 +1,460 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import {
+	allowedSwapCoinsList,
+	Coins,
+	coinsMap,
+	getUSDCurrency,
+	useBalanceConversion,
+	useSuiBalanceInUSDC,
+} from '_app/hooks/useDeepBook';
+import { useSortedCoinsByCategories } from '_app/hooks/useSortedCoinsByCategories';
+import BottomMenuLayout, { Content, Menu } from '_app/shared/bottom-menu-layout';
+import { Button } from '_app/shared/ButtonUI';
+import { Heading } from '_app/shared/heading';
+import { InputWithAction } from '_app/shared/InputWithAction';
+import { Text } from '_app/shared/text';
+import { ButtonOrLink } from '_app/shared/utils/ButtonOrLink';
+import { CoinIcon } from '_components/coin-icon';
+import { IconButton } from '_components/IconButton';
+import Loading from '_components/loading';
+import Overlay from '_components/overlay';
+import { filterAndSortTokenBalances } from '_helpers';
+import { useActiveAddress, useCoinsReFetchingConfig } from '_hooks';
+import { DescriptionItem } from '_pages/approval-request/transaction-request/DescriptionList';
+import { MaxSlippage } from '_pages/swap/MaxSlippage';
+import { QuoteAssets } from '_pages/swap/QuoteAssets';
+import { FEES_PERCENTAGE, initialValues, type FormValues } from '_pages/swap/utils';
+import { validate } from '_pages/swap/validation';
+import { useCoinMetadata, useFormatCoin } from '@mysten/core';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
+import { ArrowDown12, ArrowRight16, ChevronDown16, Refresh16 } from '@mysten/icons';
+import { MIST_PER_SUI } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import clsx from 'classnames';
+import { Form, Formik, useFormikContext } from 'formik';
+import { useMemo, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+function useCoinTypeData(activeCoinType: string | null) {
```

Comment:
Why is this name as `activeCoinType`? Does it have a special meaning? Seems that is just the type of the coin that we need to fetch the balance

Reply:
this is because of this line `const coin = coins?.find(({ coinType }) => coinType === activeCoinType);`

There would be 2 `coinType` otherwise